src/GafferImageUI/ImageView.cpp

```diff
@@ -930,8 +930,9 @@ class ImageView::ColorInspector : public Signals::Trackable
 
 	public :
 
-		ColorInspector( ImageView *view )
-			:	m_view( view ),
+		ColorInspector( Node *parent, Gadget *uiParent, ImagePlugPtr image, StringVectorDataPlugPtr channelsPlug )
```

Comment:
> This is very likely the wrong direction, but since I'm embarking on a significant refactor purely for the purpose of clarity, it felt like it made sense to explore a little.

I wouldn't say _wrong_, but it is going away from the original idea I had for this stuff, which was that they were all bolt-on components specifically for a _View_. In many ways they're like Tools, and I can envisage wanting to bolt different sets of these together for custom views in custom apps/panels (after making them into public classes at some point). I also think there's a reasonable chance that as we add more functionality to the ImageView, we'll want ColorInspector to actually be a Tool, on by default, but switch-offable to make room for other toolbars. I'm reluctant to move away from this broad approach, but I think you have some good individual points below...

> ColorInspector is less reliant on knowing the internal structure of ImageView, at the call site in ImageView seems like a better place to reference specifics of ImageView

This is a reasonable point, but for me the solution is to have ImageView provide a more solid public API that everyone can rely on. The `static_cast<ImageGadget *>( m_view->viewportGadget()->getPrimaryChild() )` is a good example - that's a terrible way of getting at the primary component of the view, so we should provide a public accessor for it.

> Having an explicit ImagePlug input feels like it makes it easier to document needing to make sure ImageView's DeepState and the DeepState's inside ImageSampler/Stats have the same hash

This is another thing I think could be provided by the ImageView API - something like a `const ImagePlug *flattenedImage()` method. That would be available to _all_ potential clients, so I could use it from a right-click menu I added to the Viewer using Python for instance. If it's hidden away at the point where ImageView constructs it's internal components, it's less useful.

> * It no longer assumes the first ImageGadget is the one it should connect to ( could be useful when it comes time for wipes )

I'm not sure this will work in practice - I imagine the constructor might get pretty unwieldy trying to communicate all the parameters for the two images and the two wipes. But the ImageView could provides an always-available API for _anyone_ to get what they need.

> The connection of the two channels plugs seems more direct and simpler than pushing the data into ImageGadget and then pulling it back out.

Agreed.